### DISCUSSION TITLE
00. PyTorch Fundamentals vector or MATRIX

### DISCUSSION THREAD

**Discussion by MajidBarzegari** [mrdbourke/pytorch-deep-learning#495] [Q&A]
![Screenshot 2023-06-21 182957](https://github.com/mrdbourke/pytorch-deep-learning/assets/17810128/85ad8a4d-43fb-4b5d-9864-9b35bbeb7698)

Hello dear friends.
  I think the tensor marked in the attached photo is related to
00. There is PyTorch Fundamentals. It is a MATRIX or a TENSOR and not a vector. what is your opinion?

**Comment by mrdbourke** [comment]
Hi @tiny-eng ,

You're right it could be considered a matrix from the point of view of it being vertical (this could be confusing).

However, since it only contains a single dimension (e.g. `[1, 2, 3]` is a vector, the same with `[34, 568, 21, 212, 311]`, even though they're different lengths, they're still in a single dimension (one pair of brackets).

In code, you'll typically only find vectors in the horizontal position (due to how they are displayed via print outs/code outputs).
